src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -82,6 +88,9 @@ protected virtual void HandleNonHttpsRequest(AuthorizationFilterContext filterCo
                     host = new HostString(host.Host);
                 }
 
+                //i use MvcOption.requireHttpsPermanent value if Permanent parameter is null
```

Comment:
Also please follow make the comment style more consistent. Make it impersonal (remove the "I") and start with a capital letter.